code/scripting/hook_conditions.cpp

```diff
@@ -158,13 +159,13 @@ static int conditionParseRawControl(const SCP_string& name) {
 
 // ---- Hook Condition Helpers ----
 
-#define HOOK_CONDITION_SHIPP(classname, documentationAddendum, shipp) \
-	HOOK_CONDITION(classname, "Ship", "Specifies the name of the ship " documentationAddendum, shipp, conditionParseString, conditionCompareShip); \
-	HOOK_CONDITION(classname, "Ship class", "Specifies the class of the ship " documentationAddendum, shipp, conditionParseShipClass, conditionCompareShipClass); \
-	HOOK_CONDITION(classname, "Ship type", "Specifies the type of the ship " documentationAddendum, shipp, conditionParseShipType, conditionCompareShipType); 
+#define HOOK_CONDITION_SHIPP(classname, prefix, documentationAddendum, shipp) \
+	HOOK_CONDITION(classname, prefix "Ship", "Specifies the name of the ship " documentationAddendum, shipp, conditionParseString, conditionCompareShip); \
+	HOOK_CONDITION(classname, prefix "Ship class", "Specifies the class of the ship " documentationAddendum, shipp, conditionParseShipClass, conditionCompareShipClass); \
+	HOOK_CONDITION(classname, prefix "Ship type", "Specifies the type of the ship " documentationAddendum, shipp, conditionParseShipType, conditionCompareShipType); 
 
-#define HOOK_CONDITION_SHIP_OBJP(classname, documentationAddendum, objp_) \
-	HOOK_CONDITION(classname, "Ship", "Specifies the name of the ship " documentationAddendum, objp_, conditionParseString, [](const object* objp, const SCP_string& shipname) -> bool { \
+#define HOOK_CONDITION_SHIP_OBJP(classname, prefix, documentationAddendum, objp_) \
```

Comment:
That prefix will need to also apply to the other two conditions generated by this macro

Reply:
Done